httpbp/client_middlewares.go

```diff
@@ -150,28 +150,27 @@ func CircuitBreaker(config breakerbp.Config) ClientMiddleware {
 			newBreaker := pool.Get()
 			breaker, loaded := breakers.LoadOrStore(host, newBreaker)
 			if loaded {
-				defer pool.Put(newBreaker)
+				pool.Put(newBreaker)
 			}
 
-			resp, err := breaker.(breakerbp.FailureRatioBreaker).Execute(func() (interface{}, error) {
-				resp, err := next.RoundTrip(req)
+			var resp *http.Response
+			_, err := breaker.(breakerbp.FailureRatioBreaker).Execute(func() (interface{}, error) {
+				var err error
+				resp, err = next.RoundTrip(req)
```

Comment:
Don't leak values out of the current scope before the error has been checked.  This has resulted in hours or days of debugging in the past, since a partial value can leak FAR from where it is created before it causes an issue, and the change that triggers it is almost always very far from where it gets detected.

```
r, err := ...
if err != nil { ... }
resp = r
```